buildSrc/src/main/kotlin/JsConfig.kt

```diff
@@ -1,33 +1,40 @@
 /*
  * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
  */
-@file:Suppress("UNUSED_VARIABLE")
 
+import internal.*
 import org.gradle.api.*
 import org.gradle.kotlin.dsl.*
 import org.jetbrains.kotlin.gradle.targets.js.ir.*
 import java.io.*
 
 fun Project.configureJs() {
-    configureJsTasks()
+    val nodeJsEnabled = project.targetIsEnabled("js.nodeJs")
+    val browserEnabled = project.targetIsEnabled("js.browser")
 
     kotlin {
+        js {
+            if (nodeJsEnabled) nodejs()
+            if (browserEnabled) browser()
+        }
+
         sourceSets {
-            val jsTest by getting {
+            jsTest {
                 dependencies {
-                    implementation(npm("puppeteer", Versions.puppeteer))
+                    implementation(npm("puppeteer", libs.versions.puppeteer.get()))
                 }
             }
         }
     }
 
+    configureJsTasks(nodeJsEnabled)
     configureJsTestTasks()
 }
 
-private fun Project.configureJsTasks() {
+private fun Project.configureJsTasks(nodeJsEnabled: Boolean) {
```

Comment:
minor: probably this could be inclined into DSL above to simplify the buildscript